rskj-core/src/main/java/co/rsk/RskContext.java

```diff
@@ -382,12 +384,21 @@ public synchronized TransactionPool getTransactionPool() {
                     getTransactionExecutorFactory(),
                     getReceivedTxSignatureCache(),
                     rskSystemProperties.txOutdatedThreshold(),
-                    rskSystemProperties.txOutdatedTimeout());
+                    rskSystemProperties.txOutdatedTimeout(),
+                    getTxQuotaChecker(),
+                    this::getWeb3); // Supplier to bypass cyclic dependency: TransactionPool needs Web3 and Web3 needs TransactionPool
```

Comment:
hmm.., `Web3` is like a frontend part of the system and passing its instance to the "core" class's ctor may lead to a system design issues in the future, as well as, having cyclic dependencies.

As I understand, you only need `GasPriceTracker` instance in the tx pool? Can we instantiate gas tracker in RskContext and share it between tx pool and web3?

Reply:
Yes, good idea, I wasn't happy with my approach either

Reply:
Done